File: GuestFlow.Application/Operations/Currency/CurrencyService.cs
using GuestFlow.Domain.Entities.Enum;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;

namespace GuestFlow.Application.Operations.Currency
{
    public class CurrencyService : ICurrencyService
    {
        private readonly IConfiguration _configuration;
        private readonly ILogger<CurrencyService> _logger;
        private readonly string _defaultCurrency;

        public CurrencyService(IConfiguration configuration, ILogger<CurrencyService> logger)
        {
            _configuration = configuration;
            _logger = logger;
            _defaultCurrency = _configuration["CurrencySettings:DefaultCurrency"] ?? "TRY";
        }

        public string GetDefaultCurrency()
        {
            return _defaultCurrency;
        }

        public bool IsValidCurrency(string currencyCode)
        {
            if (string.IsNullOrWhiteSpace(currencyCode))
                return false;

            return CurrencyHelper.IsValidCurrencyCode(currencyCode);
        }

        public string GetCurrencySymbol(string currencyCode)
        {
            if (string.IsNullOrWhiteSpace(currencyCode))
                currencyCode = _defaultCurrency;

            var currency = CurrencyHelper.FromCode(currencyCode);
            return CurrencyHelper.GetSymbol(currency);
        }

        public List<CurrencyInfo> GetSupportedCurrencies()
        {
            return new List<CurrencyInfo>
            {
                new CurrencyInfo { Code = "TRY", Name = "Türk Lirası", Symbol = "₺" },
                new CurrencyInfo { Code = "USD", Name = "Amerikan Doları", Symbol = "$" },
                new CurrencyInfo { Code = "EUR", Name = "Euro", Symbol = "€" },
                new CurrencyInfo { Code = "GBP", Name = "İngiliz Sterlini", Symbol = "£" },
                new CurrencyInfo { Code = "RUB", Name = "Rus Rublesi", Symbol = "₽" }
            };
        }
    }
}

File: GuestFlow.Frontend/src/services/currencyService.ts
import apiClient from './api'

export interface Currency {
  code: string
  name: string
  symbol: string
}

export interface CurrencySettings {
  defaultCurrency: string
  supportedCurrencies: Currency[]
}

export interface CurrencyValidation {
  isValid: boolean
  currencyCode: string
}

export interface CurrencySymbol {
  currencyCode: string
  symbol: string
}

export const currencyService = {
  getDefaultCurrency: async (): Promise<string> => {
    const response = await apiClient.get('/Currency/default')
    return response.data.data.currency
  },

  getSupportedCurrencies: async (): Promise<Currency[]> => {
    const response = await apiClient.get('/Currency/supported')
    return response.data.data
  },

  validateCurrency: async (currencyCode: string): Promise<CurrencyValidation> => {
    const response = await apiClient.get(`/Currency/validate/${currencyCode}`)
    return response.data.data
  },

  getCurrencySymbol: async (currencyCode: string): Promise<CurrencySymbol> => {
    const response = await apiClient.get(`/Currency/symbol/${currencyCode}`)
    return response.data.data
  },

  getCurrencySettings: async (): Promise<CurrencySettings> => {
    const response = await apiClient.get('/Configuration/currency')
    return response.data.data
  },
}

